src/Security/Authentication/Certificate/src/X509CertificateExtensions.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Globalization;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.AspNetCore.Authentication.Certificate
+{
+    public static class X509Certificate2Extensions
+    {
+        public static bool IsSelfSigned(this X509Certificate2 certificate)
+        {
+            return certificate.SubjectName.RawData.SequenceEqual(certificate.IssuerName.RawData);
+        }
+
+        public static string SHA256Thumprint(this X509Certificate2 certificate)
```

Comment:
If you're netcoreapp30 this is just `cert.GetCertHashString(HashAlgorithmName.SHA256)`.  (Seems weird to me that you'd be putting public extensions here)

Reply:
Nice, switched to that